benchmark/src/main/java/com/linecorp/decaton/benchmark/DecatonRunner.java

```diff
@@ -113,22 +125,69 @@ public String get(String key) {
                                                                 TaskMetadata.builder().build(), task, bytes);
                                                     })
                                          .thenProcess(
-                                                 (ctx, task) -> {
-                                                     resourceTracker.track(Thread.currentThread().getId());
-                                                     recording.process(task);
-                                                 }))
+                                                 (ctx, task) -> recording.process(task)))
                 .stateListener(state -> {
                     if (state == SubscriptionStateListener.State.RUNNING) {
                         startLatch.countDown();
                     }
                 })
                 .build();
-        resourceTracker.track(subscription.getId());
         subscription.start();
 
         startLatch.await();
     }
 
+    /**
+     * This value comes from VirtualThread#createDefaultScheduler(), which defaults keep-alive of ForkJoinPool
+     * to 30 seconds (as of Java21).
+     */
+    private static final long DRIP_INTERVAL = 30;
+    /**
+     * The intention of this method is to setup a scheduled work submitting some VirtualThread instances
+     * periodically, in order to make sure the {@link ForkJoinPool} to keep-alive the created carrier thread
+     * during warmup phase.
+     * It is necessary because {@link #onWarmupComplete(ResourceTracker)} depends on the alive list of
+     * ForkJoinPool threads to observe cpu time and memory allocation profile, which cannot be obtained from
+     * VirtualThread instances directly.
+     */
+    private void maybeSetupForkJoinPoolDrip() {
+        if (subPartitionRuntime != SubPartitionRuntime.VIRTUAL_THREAD) {
+            return;
+        }
+        ScheduledExecutorService executor = Executors.newScheduledThreadPool(
+                1, Thread.ofPlatform().daemon().factory());
+        executor.scheduleAtFixedRate(() -> {
+            for (int i = 0; i < Runtime.getRuntime().availableProcessors(); i++) {
```

Comment:
Does submitting tasks up to `availableProcessors` count ensure keeping all ForkJoinPool worker thread to keep alive?

Reply:
Yeah it worked based on my observation.